### PR TITLE
Feat/add e2e for chat

### PR DESCRIPTION
**Pull request by Jamerrone**
- [Trello](https://trello.com/c/38f377MQ/108-5-add-e2e-for-chat)

### CODE REVIEW

File: src/client/components/lazy-chat/lazy-chat.vue [hunk 1]
```diff
@@ -12,6 +12,7 @@
       class="button button--primary lazy-chat__open-button"
       :class="{ 'lazy-chat__open-button--pending': (isAccepted && !isLoaded) }"
       @click="handleClick"
+      v-test:lazyChatButtonIdentity
```
Comment (jbmoelker) [inline src/client/components/lazy-chat/lazy-chat.vue]: The reason we had "Identity" in the selector name is that the element around the logo was simply called "brand identity". You don't have to use it in every test directive. So simply `lazyChatButton` etc.

File: test/cypress/integration/lazy-chat.test.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import { crispWebsiteId } from '../../../src/client/static/data/app.json'
```
Comment (jbmoelker) [inline test/cypress/integration/lazy-chat.test.js:1]: can you use a variable here so other developers know what the outcome means?
```js
const chatIsEnabled = !!crispWebsiteId
```

File: test/cypress/integration/lazy-chat.test.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+const lazyChatButtonIdentity = '[data-test="lazyChatButtonIdentity"]'
+const lazyChatOptInIdentity = '[data-test="lazyChatOptInIdentity"]'
+
+describe('Lazy chat', () => {
+  beforeEach(() => {
+    cy.visit('/')
+  })
+
+  it('is only present if Crisp ID is set', () => {
```
Comment (jbmoelker) [inline test/cypress/integration/lazy-chat.test.js]: Use a variable so other devs directly see both the if and the else apply to the same element
```js
const button = cy.get(lazyChatButton)
if (chatIsEnabled) {
  button.should('be.visible')
} else {
  button.should('not.exist')
}
```

File: test/cypress/integration/lazy-chat.test.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    if (!!crispWebsiteId) {
+      cy.get(lazyChatButtonIdentity).should('be.visible')
+    } else {
+      cy.get(lazyChatButtonIdentity).should('not.exist')
+    }
+  })
+
```
Comment (jbmoelker) [inline test/cypress/integration/lazy-chat.test.js]: I think you need to make all of the tests after this point conditional. They only apply if the chat is enabled, so:
```js
if (!chatIsEnabled) {
  return
}
```

File: test/cypress/integration/lazy-chat.test.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  it('asked me to accept the T&C when I hit the chat button ("terms_conditions" === false)', () => {
+    cy.clearLocalStorage()
```
Comment (jbmoelker) [inline test/cypress/integration/lazy-chat.test.js]: I think Cypress allows you to be more specific, so you don't affect other tests:
```js
cy.clearLocalStorage('terms_conditions')
```

File: test/cypress/integration/lazy-chat.test.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+  it('is only present if Crisp ID is set', () => {
+    if (!!crispWebsiteId) {
+      cy.get(lazyChatButtonIdentity).should('be.visible')
+    } else {
+      cy.get(lazyChatButtonIdentity).should('not.exist')
+    }
+  })
+
+  it('asked me to accept the T&C when I hit the chat button ("terms_conditions" === false)', () => {
```
Comment (jbmoelker) [inline test/cypress/integration/lazy-chat.test.js]: Try to only use human language here, so something like:
"when I haven't accepted T&C yet, I'm prompted to accept the T&C when I hit the chat button"

(same for the following test)

File: test/cypress/integration/lazy-chat.test.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
+import { crispWebsiteId } from '../../../src/client/static/data/app.json'
+
+const lazyChatButtonIdentity = '[data-test="lazyChatButtonIdentity"]'
```
Comment (jbmoelker) [inline test/cypress/integration/lazy-chat.test.js]: everything here is scoped to this file and we're already in the `lazy-chat` directory, so you can simplify your variable names to be more readable (`chatButton`, `optInDialog` (or whatever it is))

File: test/cypress/integration/lazy-chat.test.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  it('dit not ask me to accept the T&C when I hit the chat button ("terms_conditions" === true)', () => {
+    localStorage.setItem('terms_conditions', true);
+    cy.get(lazyChatButtonIdentity).click()
+    cy.get(lazyChatOptInIdentity).should('not.exist')
+  })
+
+  it('is only present if T&C is accepted', () => {
+    cy.get(lazyChatButtonIdentity).click()
+    cy.get(`${lazyChatOptInIdentity} .opt-in__buttons`)
+      .children('button.button--primary')
```
Comment (jbmoelker) [inline test/cypress/integration/lazy-chat.test.js]: add a `v-test` to the button you want to address here (same for following test)

File: test/cypress/integration/lazy-chat.test.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    cy.get(lazyChatOptInIdentity).should('be.visible')
+  })
+
+  it('dit not ask me to accept the T&C when I hit the chat button ("terms_conditions" === true)', () => {
+    localStorage.setItem('terms_conditions', true);
+    cy.get(lazyChatButtonIdentity).click()
+    cy.get(lazyChatOptInIdentity).should('not.exist')
+  })
+
+  it('is only present if T&C is accepted', () => {
```
Comment (jbmoelker) [inline test/cypress/integration/lazy-chat.test.js]: `it('Crisp client is only ...`

File: test/cypress/integration/lazy-chat.test.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+    cy.get(lazyChatButtonIdentity).click()
+    cy.get(`${lazyChatOptInIdentity} .opt-in__buttons`)
+      .children('button.button--primary')
+      .click()
+    cy.get('.crisp-client').should('exist')
+  })
+
+  it('setted the focus correctly', () => {
```
Comment (jbmoelker) [inline test/cypress/integration/lazy-chat.test.js]: `it('sets the focus on the message input when opening the Crisp client'`

**Review by jbmoelker — CHANGES_REQUESTED** [review]
Please also add new tests for or add to existing tests:
* focus is on "agree" button when opt-in is opened
* opt-in disappears on "decline"
* focus is on chat button after "decline"

**Review by vbvnvmfjg — APPROVED** [review]

**Review by jbmoelker — APPROVED** [review]

**Pull request merged**